Documentation/ceph-cluster-crd.md

```diff
@@ -1489,3 +1490,42 @@ data:
 
 Note: if you are using self-signed certificates (not known/approved by a proper CA) you must pass `VAULT_SKIP_VERIFY: true`.
 Communications will remain encrypted but the validity of the certificate will not be verified.
+
+##### Ceph CSI configuration
+
+Ceph-CSI is capable of encrypting requested PVs.
+Rook can configure Ceph-CSI to turn on encryption with Key Management System support.
+For this, you need to add new parameter in your CephCluster CR spec:
+
+```yaml
+security:
+  kms:
+  ...
+  ...
+  csiConnectionDetailsCMName: "rook-ceph-csi-vault"
+```
+
+The config map will look like:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: rook-ceph-csi-vault
+  namespace: rook-ceph
+data:
+  csiConfig: |-
+    {
+      "vault-test": {
+        "encryptionKMSType": "vault",
+        "vaultAddress": "http://vault.default.svc.cluster.local:8200",
+        "vaultAuthPath": "/v1/auth/kubernetes/login",
+        "vaultRole": "csi-kubernetes",
+        "vaultPassphraseRoot": "/v1/secret",
+        "vaultPassphrasePath": "ceph-csi/",
```

Comment:
The vaultAuthPath, vaultPassphraseRoot and vaultPassphrasePath are currently not used by the VaultTokensKMS implementation. These can surely be added, but ideally the options would match the standard Vault environment variables (like `vaultAddress` = `VAULT_ADDRESS`), but that will be a TODO for the time  being.

Are there any options that you expect to need to configure? I'll add them on the Ceph-CSI side of things then.